src/main/java/com/google/codeu/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,81 @@
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.util.List;
+
+import java.util.Map;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload
+ * and then forwards the request to this servlet. This servlet can then
+ * process the request using the file URL we get from Blobstore.
+ */
+@WebServlet("/my-form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // Get the message entered by the user.
+        String foodName = request.getParameter("foodName");
+        String amount = request.getParameter("amount");
+        String date = request.getParameter("date");
+
+        // Get the URL of the image that the user uploaded to Blobstore.
+        String imageUrl = getUploadedFileUrl(request, "image");
+
+        // Output some HTML that shows the data the user entered.
+        // TODO store in Datastore instead of just printing out
+        ServletOutputStream out = response.getOutputStream();
+        out.println("<p>Here's the "  + foodName + " you uploaded</p>");
+        out.println("<a href=\"" + imageUrl + "\">");
+        out.println("<img src=\"" + imageUrl + "\" />");
+        out.println("</a>");
+        out.println("<p>You ate " + amount + " of this on " + date + ".</p>");
+    }
+
+    /**
+     * Returns a URL that points to the uploaded file, or null if the user didn't upload a file.
+     */
+    private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName){
+        BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+        List<BlobKey> blobKeys = blobs.get("image");
+
+        // User submitted form without selecting a file, so we can't get a URL. (devserver)
+        if(blobKeys == null || blobKeys.isEmpty()) {
+            return null;
+        }
+
+        // Our form only contains a single file input, so get the first index.
+        BlobKey blobKey = blobKeys.get(0);
+
+        // User submitted form without selecting a file, so we can't get a URL. (live server)
+        BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+        if (blobInfo.getSize() == 0) {
+            blobstoreService.delete(blobKey);
+            return null;
+        }
+
+        // We could check the validity of the file here, e.g. to make sure it's an image file
+        // https://stackoverflow.com/q/10779564/873165
```

Comment:
This is definitely worth doing. Otherwise you run the risk of people uploading all sorts of bad stuff.